### PR TITLE
Cluster-autoscaler: do full scheduling check in calculating node utilization in scale-down

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @piosz @fgrzadkowski

### CODE REVIEW

File: cluster-autoscaler/simulator/cluster.go [hunk 1]
```diff
@@ -29,40 +29,53 @@ import (
 	"github.com/golang/glog"
 )
 
-// FindNodeToRemove finds a node that can be removed.
-func FindNodeToRemove(candidates []*kube_api.Node, allNodes []*kube_api.Node, pods []*kube_api.Pod,
-	client *kube_client.Client, predicateChecker *PredicateChecker) (*kube_api.Node, error) {
+// FindNodesToRemove finds a node that can be removed.
```
Comment (piosz) [inline cluster-autoscaler/simulator/cluster.go]: Please fix the comment

File: cluster-autoscaler/simulator/cluster.go [hunk 2]
```diff
@@ -29,40 +29,53 @@ import (
...
+func FindNodesToRemove(candidates []*kube_api.Node, allNodes []*kube_api.Node, pods []*kube_api.Pod,
+	client *kube_client.Client, predicateChecker *PredicateChecker, maxCount int,
+	fastCheck bool) ([]*kube_api.Node, error) {
 
 	nodeNameToNodeInfo := schedulercache.CreateNodeNameToInfoMap(pods)
```
Comment (piosz) [inline cluster-autoscaler/simulator/cluster.go]: Please reuse here https://github.com/kubernetes/contrib/blob/master/cluster-autoscaler/utils.go#L207

**Comment by piosz** [comment]
LGTM, just two nits

File: cluster-autoscaler/simulator/cluster.go (hunk 2 above)
Comment (mwielgus) [inline cluster-autoscaler/simulator/cluster.go]: Next pr

File: cluster-autoscaler/simulator/cluster.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/simulator/cluster.go]: Done.

**Pull request merged**